### DISCUSSION TITLE
Quais os melhores cursos online para desenvolvedores?

### DISCUSSION THREAD

**Discussion by djteix** [frontendbr/forum#1515] [Discussão]
Olá a todos, estou começando essa caminhada para me tornar um desenvolvedor, estou me guiando pela curiosidade, no momentos estou estudando, HTML5, CSS3, JS e NodeJS. Não está fácil! Vi algumas plataformas online de cursos como udemy, alura, etc. Poderiam me indicar as melhores em metodologia e que os certificados tenham algum respeito no mercado. Obrigado.

**Comment by yuritoledo** [comment]
Cara, dá uma olhada no site do freecodecamp.org

Lá tem tudo o que você precisa pra começar, como Css html, js, bootstrap, node, react, algoritmos e mais um monte de coisa

Fora que tem uns desafios pra você praticar 

E tem uns certificados também!

**Comment by renedocket** [comment]
Olá djteix, tudo bem? 

- **Rocketseat** - https://rocketseat.com.br/starter)
- **DevPleno** - https://www.devpleno.com/
- **Collabcode** - https://collabcode.training/ o curso deles é voltado para front-end, que além de abordar o básico HTML5 / CSS3 / JS tem mais alguns adicionais, acompanhe pela comunidade que fica mais fácil de saber quando abre as inscrições. 
- **Alura** é bem interessante eu aprendi muito com os cursos básicos deles, e consegui um trampo por meio desses conhecimentos, acredito que é uma boa se manter nele.

Espero ter ajudado! :)

**Comment by ThiagoHs1** [comment]
A **alura** parece ser muito bom, planejo comprar quando tiver um desconto

Recentemente comprei o **pacote full stack da danki code** recomendo bastante, to aprendendo um monte de coisa, ele dá um monte de projetos para tu aprender junto enquanto faz - https://cursos.dankicode.com/

Outra plataforma de cursos que vejo as pessoas recomendarem e sendo bem falado é a **origamid**, acho que vale a pena dar uma olhada ---> https://origamid.com/planos/

**Reply by diegobarcelos** [reply]
Olá, tenho Origamid, vale muito a pena, e durante o ano ele lança promoção do plano vitalício, pagando menos de 400 reais parcelado em 12x sem juros, assim fica com todos os cursos e os próximos que aparecerem sem precisar investir mais 1 real na plataforma. A didática dele é muito boa, não fiz todos os cursos, o que fiz até o final foi o de Java Script Completo ES6, um dos melhores cursos de JS e os outros vi por cima e são muito bons! Os cursos são voltados pro Front End, mas você acaba saindo com um conhecimento bem sólido, e lá tem cursos de ReactJS e Vue.JS

**Comment by mdandrade** [comment]
Só uma dica no JavaScript. Pode parecer muito difícil no começo pra quem não é da área de programação, mas tenha há paciência que vai ficar fácil depois.

Minha caminhada foi assim:
- w3chools
- freeCodeCamp
- Origamid

Paciência e constância que vai.

**Comment by viniciusreiss** [comment]
Cara eu assinei alura por 1 ano, olha os cursos tem uma qualidade boa, porém alguns estavam muito desatualizados e com uma didática ruim como o de React por exemplo. A udemy é uma execelente opçāo pois os cursos sāo baratos, recomendo olhar os da Cod3R.

**Comment by TL-Developer** [comment]
Udemy também é uma boa hein ;), mas comecei pela Alura.

**Comment by alecrs** [comment]
Sei que o assunto já tem um mês de forum, mas acho importante a diversidade de opiniões, então deixo como dica a plataforma de estudo que uso até hoje: Devmedia. Conteúdo bom, completo e orientado para que possa ser estudado passo a passo, do básico ao avançado.
www.devmedia.com.br

**Comment by jpferreiradev** [comment]
Pessoal, tudo bom com vocês? 

Tiveram pessoas aqui que assinaram a Alura, e eu assinei também só que o seguinte, eu 
não consegui aprender com a metodologia deles para o curso básico de HTML E CSS, ele é dividido 
em duas partes, nessa primeira parte começa com um pouco de Html e depois é só CSS, mesmo 
estudando todo dia, sinto que não conseguir aprender um pouco pelo menos, terminei a primeira parte do curso hoje, eu não sei se o problema sou eu, kkkkk! O que vocês acharam do curso deles?

**Comment by willycamargo** [comment]
Fala @jotepauder , tudo certo?

Minha namorada também assinou Alura e sente dificuldades com a metodologia no mesmo curso. Para complementar eu tenho ensinado algumas coisas para ela e tento esclarecer os assuntos que ela fica confusa.

Em contrapartida, caso você não tenha problemas com inglês, parece que a metodologia do freecodecamp é muito boa.

Não muito atual, mas os cursos do "curso em vídeo" do Gustavo Guanabara também tem uma didática bem objetiva e foi muito bom para alguns amigos meus.

Além desses com certeza existem outros que não conheço. Mas ficam as dicas.

Abrs e continue estudando.

**Reply by danielmlima1971** [reply]
Eu fiz do Gustavo Guanabara e é muito top. Recomendo muito, até pra quem já sabe e quer relembrar alguns detalhes ou se aprimorar.

**Reply by Josiel-Sousa** [reply]
didatica objetiva? do guanabara? to no curso de JS errado :/

**Reply by LucasTroglio** [reply]
Estou fazendo Phyton  com ele, top

**Comment by jpferreiradev** [comment]
Tudo sim @willycamargo , e com você?

Ah, que bom em pensar que não sou eu que sinto isso, kkkk! Eu tenho até começado o 
freecodecamp, mas assinei a Alura mais por isso, o meu inglês ainda não é muito bom. E vejo que tem bastante material em inglês muito bom. Eu fiquei em migrar pelo menos essa parte para o curso em Video, ali eu consigo entender e muito bem diversos assuntos que o Gustavo Guanabra fala, é muito foda, eu acho que pra lá e depois eu volto para Alura, mesmo assim muito obrigado pela ajuda o//

Abraço, continuarei sim o/

**Comment by ninetails** [comment]
Respondendo a pergunta original, o melhor (e já que não perguntou de preço) que ouvi falar, das gringa (inglês somente) é o **Frontend Masters**. Tem só cara pica ministrando curso.

**Reply by Kaedh** [reply]
Inclusive o Github tem um pacote de ferramentas e cursos para quem é estudante, e lá tem uma licença de 6 meses grátis para o 
[Frontend  Masters](https://frontendmasters.com/welcome/github-student-developers/).

**Comment by 45lailson** [comment]
Udemy tem cursos excelentes tbm

**Comment by dheysonalves** [comment]
@djteix Acredito que depende da didática de cada professor, mas como se refere a curso de forma geral. Existem alguns que possuem uma boa grade, que ajuda você a progredir, com um fluxo bem definido, segue alguns que passei em algum momento e recomendo bastante:
- [Free Code Camp](https://www.freecodecamp.org/)
- [Front End Masters](https://frontendmasters.com/)
Esse aqui, apenas possui preço em dolar. Mas o programa do [Github Developer Pack](https://education.github.com/pack), presenteia com 1 ano de licença, foi assim que consegui acesso na época.
- [Developer Roadmap](https://github.com/kamranahmedse/developer-roadmap)
Você pode também buscar criar sua própria grade de estudos, seguindo esse Roadmap, ou colocar no fluxo de estudos como complemento. Na época, ajudou bastante no meu início, recomendo dá uma olhada.

Espero ter ajudado 😄 ✋🏾

**Reply by Hersonmei** [reply]
Valeu pelas dicas!! 

Consegui o acesso do Frontend Masters aqui pelo github e estou achando bem interessante o curso. Tu tem alguma dica de como tirar um maior proveito de lá? Você seguiu os paths proposto por eles?

Valeu!

**Reply by Maicon00001** [reply]
Voce conseguiu ingressa na area? Com base no curso do masters

**Comment by wilsonneto-swe** [comment]
Caso tenha preferencia por cursos BR, dois que fiz quando estava estudando React e me ajudaram demais foram:
Dev JS da One Bit Code (acho que foi R$ 297,00)
http://bit.ly/onebitcode-fullstack-js

E o curso focado em ReactJS do Dev Pleno também gostei bastante:
(porém acho que foi R$ 697,00, ambos tem um custo benefício muito bom em minha opinião)
http://bit.ly/react-dev-pleno

Espero que ajude, forte abraço!

**Reply by camillageber** [reply]
@wilsonneto-dev , estava pesquisando informações sobre esse curso da One Bit Code e achei seu comentário aqui... Você pode me falar um pouco mais sobre ele? Para quem é totalmente inciante, vale a pena? 
Se não se importa em dizer, você já trabalha na área? 

Agradeço!

**Comment by theandersonn** [comment]
Contribuindo com algumas indicações.

Esses aqui fiz ano passado e me surpreenderam pela qualidade.

**Advanced CSS and Sass: Flexbox, Grid, Animations**
https://www.udemy.com/course/advanced-css-and-sass/

**Aprenda o novo React com Hooks criando 8 projetos práticos**
https://www.udemy.com/course/aprenda-react-hooks-8-projetos-praticos/

**React Tutorial and Projects Course**
https://www.udemy.com/course/react-tutorial-and-projects-course/

Estou fazendo esse aqui agora:

**React Avançado: Crie aplicações com NextJS, GraphQL**
https://www.udemy.com/course/react-avancado/

Próximo curso que farei:

**Aprenda a testar Aplicações Javascript**
https://javascript.tv.br/

_Estou pesquisando por cursos sobre a AWS, se tiverem indicações, compartilhem._

**Comment by Hersonmei** [comment]
Eu tô iniciando na área, estou fazendo o curso do ProgramadorBR e fazendo o freecodecamp junto pra ir fundamentando. Como não sou da área, é importante ir testando algumas metodologias diferentes para ver qual funciona melhor!

Bons estudos!

**Reply by RafaelFerraz** [reply]
Oi. Estou pensando em comprar o curso do ProgramadorBR, vi boas recomendações por aí. Já tem um tempo que você está nele né? Recomenda?

**Reply by Hersonmei** [reply]
@RafaelFerraz Eu dei uma parada lá, mas o curso realmente é bem avaliado. 
O curso cobre muita coisa, acho que dá pra ir construindo uma base boa lá.
Eu não consegui prosseguir no curso pq acabei percebendo que meu dou melhor em estudo por livro, no meu ritmo, sem precisar ficar parando toda hora o vídeo. E também acabei não identificando na parte Front-end, apesar do curso ser Full Stack, mas como não sou da área foi uma tentativa do primeiro contato com o mundo da programação.

Mesmo assim não consegui de deixar continuar com interesse na área, por isso voltei aos estudos com a linguagem Python na forma de livro desde abril, e tô gostando bastante, acabei de identificado mais por aqui. E se um dia for para desenvolvimento web com Python posso voltar lá no curso do programadorBR para relembrar a base de HTML, CSS e JS. E a base de lógica de programação que peguei lá já ajudou meu início em Python ser ainda mais tranquilo.

É isso, não sei se consegui ajudar, mas eu parei mesmo por não identificar muito com a área de front e por fazer um curso de programação em vídeo.  Mas lá se não me engano, você tem um período de 7 dias de teste. 

Acho que já deve ter visto esse vídeo, mas se não, esse cara aí começou no curso dele depois aprofundou com outros cursos e tá seguindo bem na profissão: https://www.youtube.com/watch?v=W1U5lz1aQWY&ab_channel=ARTHURPC

Boa sorte aí e bons estudos!

**Reply by RafaelFerraz** [reply]
Obrigado pela resposta, ajudou bastante sim. Eu vi o vídeo que você indicou. Eu to iniciando, estou fazendo alguns cursos na Digital Innovation One. Acabei de comprar o curso do ProgramadorBR, esses 7 dias de garantia serão uma boa proteção caso eu não me adapte com o curso.

Muito obrigado, ótimo dia pra você 🙏🏻😊

**Reply by Hersonmei** [reply]
E aí Rafael, deu certo o curso? 
Cara, fiquei meio perdido lá em Python e decidi dar mais uma chance para o front end, estou iniciando um curso recomendado aqui pelo @dheysonalves e estou achando bem interessante, Frontend Masters.

Acho que vou ficar usando o ProgramadorBr como um suporte, uma segunda fonte de consulta e para tirar dúvidas tbm, já que nesse curso que estou fazendo eles não tem um suporte direto com os alunos.

**Comment by LeonardoAtripoli** [comment]
Oque acham do full stack da impacta??? eu fiz alguns cursos la ....linux...oracle...SQL server...e todos me trouxeram muito resultado e todos os professores eram excelentes ...mas isso foi a uns 7 anos quando terminei o ultimo curso que fiz....hoje por ser tudo LiveClass não sei se seria uma boa .. o valor é bem alto do curso.....mas....... As demais vezes que apostei todas foram positivas e me trouxeram oportunidades de emprego simplesmente por ter concluído o curso. (esta em média de 12 mil reais o curso).

**Comment by fulviavaz** [comment]
Eu queria mto assinar Alura eles podiam colocar a opção assinatura mensal pra galera q n tem cartão

**Reply by AllancGoulart** [reply]
Tbm queria esse método de pagamento, é uma pena ate para quem tem cartão comprometer os 900 reais de limite em uma pancada só;

**Reply by emilainesantos** [reply]
Oi, eles tem a opção no boleto

**Comment by wilsonneto-swe** [comment]
Fiz um repo há tempo atrás que coloquei alguns dos cursos que já fiz e achei muito massa:
https://github.com/wilsonneto-dev/Melhores_Cursos_Online

**Comment by Wanderson-png** [comment]
Olá amigos, queria saber se é bom este curso do Senac de Desenvolvedor Full-stack, ou da Digital House para quem quer começar nesta área? Pois a minha cidade é pequena e estou interessado em cursos com aulas 100% ao vivo. 

Obs: Terminei a faculdade de  tecnólogo em ADS a 7 anos atrás, e não segui essa área, só que sempre gostei da parte envolvendo web e banco de dados.

https://www.sp.senac.br/cursos-livres/curso-de-desenvolvedor-full-stack

**Comment by sarahdialmeida** [comment]
Oi Josiel,
Os melhores, para mim, são os do FreeCodeCamp e RocketSeat.

Bons estudos!
S.

Em qua., 18 de ago. de 2021 às 17:00, Josiel Sousa ***@***.***>
escreveu:

**Comment by brayanmonteiroo** [comment]
Cara, meu site pode te dá um norte, tem coisa paga e gratuita: https://programadoresdepre.com.br/

Mas pra ser sincero, gratuitamente e um dos melhores lugares para INICIANTES é o Curso em vídeo: https://www.cursoemvideo.com/

**Comment by vinyand** [comment]
Olá, o rockseat tem mais de 100h de curso GRATUITO entra lá a galera que quer começar nessa area tu vai fazer projetos e exercicios, pode complementar com outros cursos também e agregar.

**Reply by fabiohscampos** [reply]
Acabei de comecar la... a didatica deles eh muito boa! Sou da area de infra e quero migrar pra dev...

**Comment by volksec** [comment]
Curso em vídeo seria ótimo para iniciar.

**Comment by jonasmarco** [comment]
Alguém teria indicação de curso para quem já está a algum tempo na área, já é pleno e quer ser especialista/sênior, frontend

**Reply by willycamargo** [reply]
Se inglês não for uma barreira eu recomendo o [Frontend Masters](https://frontendmasters.com/), tem bastante conteúdo aprofundado por lá.

**Reply by jonasmarco** [reply]
Valeu pela indicação mano, só o dollar que me barra um pouco kakaka. Também descobri esse site aqui https://egghead.io/, indicação de um colega de trabalho, tem bastante conteúdo free.

**Comment by wilsonneto-swe** [comment]
Sobre o tema, escrevi um artigo com algumas indicações de cursos que já fiz e são ótimos:

https://www.javascriptnews.com.br/post/melhores-cursos-de-js-react-node-e-typescript

Espero que ajude!

**Comment by Elisaudo** [comment]
Tem o curso de programação Full Stack Javascript da 3way Academy. O curso te prepara para ser um programador junior. No curso são feitos vários projetos reais, alinhados com o mercado. 

https://www.3way.com.br/curso-de-programacao-full-stack/

**Comment by Glaydsonbusiness** [comment]
Senhores Licença, sei que já faz um tempo... Mas não vi ninguem comentando sobre a plataforma "b7web" 👇🏾
https://lp.b7web.com.br/fullstack

E me pareceu muito bom a plataforma, didática e organização, principalmente para iniciantes como eu...alguem que já fez, ou indica? pode opinar se vale a pena?

**Reply by dmchiodini** [reply]
Eu usei muito e recomendo, principalmente para iniciantes.

**Comment by wpbarcelos** [comment]
Harvard CS50, https://www.youtube.com/watch?v=LfaMVlDaQ24

**Comment by danterkive** [comment]
> Olá a todos, estou começando essa caminhada para me tornar um desenvolvedor, estou me guiando pela curiosidade, no momentos estou estudando, HTML5, CSS3, JS e NodeJS. Não está fácil! Vi algumas plataformas online de cursos como udemy, alura, etc. Poderiam me indicar as melhores em metodologia e que os certificados tenham algum respeito no mercado. Obrigado.

recomendo primeiro aprender os fundamentos : algoritmo e logica de programação, o do nelio alves da udemy é muito bom, recomendo demais
git
banco de dados
programação orientada a objetos
estrutura de dados e algoritmos
ingles

e aí escolher back end ou front end /// talvez os dois

se aprofundar bem em uma linguagem

tente ler livros (entendendo algoritmos livro do Aditya Y. Bhargava - é muito bom), participar de grupos, eventos, foruns como este, ler artigos e blogs

não tente aprender tudo de uma vez, tu só vai virar o famoso dev pato
